src/modules/dashboard/components/Dialogs/AddExistingSafeDialog/AddExistingSafeDialogForm.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { FormikProps } from 'formik';
+import { defineMessages } from 'react-intl';
+import Button from '~core/Button';
+import { ActionDialogProps } from '~core/Dialog';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Select, Input, Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import { chains } from './gnosisSafeChains';
+
+import { FormValues } from './AddExistingSafeDialog';
+
+import styles from './AddExistingSafeDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.title',
+    defaultMessage: 'Add Gnosis Safe',
+  },
+  chain: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.chain',
+    defaultMessage: 'Select Chain',
+  },
+  safeName: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.safeName',
+    defaultMessage: 'Name the safe',
+  },
+  annotation: {
+    id:
+      'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.annotation',
+    defaultMessage: 'Explain why you’re adding this Gnosis Safe (optional)',
+  },
+  contract: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.contract',
+    defaultMessage: 'Add Safe Contract',
+  },
+  safeFound: {
+    id:
+      'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.safeFound',
+    // @TODO remove hardcoding here
+    defaultMessage: 'Safe found on Gnosis Chain',
+  },
+});
+interface Props extends ActionDialogProps {
+  ethDomainId?: number;
+}
+
+const AddExistingSafeDialogForm = ({
+  back,
+  handleSubmit,
+  isSubmitting,
+  isValid,
+}: Props & FormikProps<FormValues>) => {
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'sidePadding' }}>
+        <div className={styles.headingContainer}>
+          <Heading
+            appearance={{ size: 'medium', margin: 'none', theme: 'dark' }}
+            text={MSG.title}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <div className={styles.chainSelect}>
+          <Select
+            options={chains}
+            label={MSG.chain}
+            name="chainId"
+            appearance={{ theme: 'grey', width: 'fluid' }}
+            disabled={isSubmitting}
+          />
```

Comment:
As previously discussed with @arrenv this should no longer be the case, as the chain should be selected automatically _(via Gnosis's API)_ based on the safe you add, rather then the other way around.

This should just be used for display purposes, and not allow the user to make any changes.

Reply:
The specs are not updated yet. @arrenv can you please maybe just tell me what should the first field look like instead?

Reply:
@dit7ya @rdig I looked into this more and tested the API, you definitely need to have the chain selected first. Unless you know how to do this. But the chain selection is required in order to select the correct transaction service API endpoint.

So, the design should still be the same.

Reply:
@rdig @arrenv Please let me know what to do about it.

Reply:
@rdig I added these in to the networks and added a new typing to the `NetworkInfo`, it is questionable, however reduces repetition.

Reply:
I don't think it's questionable. I would have wrote the `GNOSIS_SAFE_NETWORKS` array a bit differently, but other than that, it's good to go.

And as long as that array format works for your use case, sure, leave it as it is.